dotnet/src/InternalUtilities/src/System/TypeConverterFactory.cs

```diff
@@ -51,6 +53,21 @@ internal static class TypeConverterFactory
             return converter;
         }
 
-        return null;
+        return new JsonSerializationTypeConverter();
+    }
+
+    internal sealed class JsonSerializationTypeConverter : TypeConverter
+    {
+        public override bool CanConvertFrom(ITypeDescriptorContext? context, Type sourceType) => true;
+
+        public override object? ConvertFrom(ITypeDescriptorContext? context, CultureInfo? culture, object value)
+        {
+            return JsonSerializer.Deserialize<object>((string)value);
```

Comment:
Does specifying `<object>` here work? How does the serializer know the actual destination type to use?

Reply:
no it doesn't. updated to pass the correct type as a 2nd argument to the `JsonSerializer`